core/src/main/java/org/axonframework/commandhandling/model/AggregateLifecycle.java

```diff
@@ -127,17 +129,14 @@ public static void markDeleted() {
      * @return the {@link AggregateLifecycle} for the current aggregate
      */
     protected static AggregateLifecycle getInstance() {
-        AggregateLifecycle instance = CURRENT.get();
-        if (instance == null && CurrentUnitOfWork.isStarted()) {
+        AggregateLifecycle instance = Scope.getCurrentScope();
+        if (CurrentUnitOfWork.isStarted()) {
```

Comment:
This is not semantically correct anymore. Aggregate instances should only be looked up in the Unit Of Work when there is no current scope. If there is an active scope, that should always be used.

Reply:
Fair point, I'll adjust it to immediately return the `Scope.getCurrentScope()` call.